test/spec/label.js

```diff
@@ -34,4 +53,329 @@ describe("label", function () {
+              name: "Null Island",
+            })
+          )
+      ).to.be.eql("");
+    });
+    it("localizes to preferred language", function () {
+      expect(
+        evaluatedExpression(["en"], {
+          "name:en": "Null Island",
+          name: "Insula Nullius",
+        })
+      ).to.be.eql("Null Island");
+    });
+  });
+
+  describe("#localizedNameWithLocalGloss", function () {
+    let evaluatedExpression = (locales, properties) =>
+      expression
+        .createExpression(
+          localizedTextField([...Label.localizedNameWithLocalGloss], locales)
+        )
+        .value.expression.evaluate(expressionContext(properties));
+
+    let evaluatedLabelAndGloss = (locales, properties) => {
+      let evaluated = evaluatedExpression(locales, properties);
+      if (typeof evaluated === "string") {
+        return [evaluated, null];
+      }
+      return [evaluated.sections[0].text, evaluated.sections[4].text];
+    };
+
+    let expectGloss = (
+      locale,
+      localized,
+      local,
+      expectedLabel,
+      expectedGloss
+    ) => {
+      let properties = {
+        name: local,
+      };
+      properties[`name:${locale}`] = localized;
+      expect(evaluatedLabelAndGloss([locale], properties)).to.be.deep.equal([
+        expectedLabel,
+        expectedGloss,
+      ]);
+    };
+
+    it("puts an unlocalized name by itself", function () {
+      expect(
+        evaluatedExpression(["en"], {
+          name: "Null Island",
+        })
+      ).to.be.eql("Null Island");
+    });
+    it("glosses an anglicized name with the local name", function () {
+      let evaluated = evaluatedExpression(["en"], {
+        "name:en": "Null Island",
+        name: "Insula Nullius",
+      });
+
+      expect(evaluated.sections.length).to.be.eql(7);
+      expect(evaluated.sections[0].text).to.be.eql("Null Island");
+      expect(evaluated.sections[1].text).to.be.eql("\n");
+      expect(evaluated.sections[2].text).to.be.eql("(\u200B");
+      expect(evaluated.sections[3].text).to.be.eql("Null Island"[0] + " ");
+      expect(evaluated.sections[4].text).to.be.eql("Insula Nullius");
+      expect(evaluated.sections[5].text).to.be.eql(" " + "Null Island"[0]);
+      expect(evaluated.sections[6].text).to.be.eql("\u200B)");
+
+      expect(evaluated.sections[3].scale).to.be.below(0.1);
+      expect(evaluated.sections[4].scale).to.be.below(1);
+      expect(evaluated.sections[5].scale).to.be.below(0.1);
+    });
+    it("deduplicates matching anglicized and local names", function () {
+      expectGloss("en", "Null Island", "Null Island", "Null Island", null);
+      expectGloss("en", "Null Island", "NULL Island", "Null Island", null);
+      expectGloss("en", "Montreal", "Montréal", "Montréal", null);
+      expectGloss("en", "Quebec City", "Québec", "Québec City", null);
+      expectGloss("en", "Da Nang", "Đà Nẵng", "Đà Nẵng", null);
+      expectGloss("en", "Nūll Island", "Ñüłl Íşlåńđ", "Ñüłl Íşlåńđ", null);
```

Comment:
I’ve implemented a compact syntax for testing that, given a name in the preferred language and a name in the local language both in the properties of a feature, the massive expression will evaluate to a `format` expression with the expected primary label and gloss. In fact, just about any expression in this style can now be tested in this manner, independently of the style, by providing the necessary bits of data.